adapters/AppLovin/AppLovinAdapter/GADMAdapterAppLovinUtils.m

```diff
@@ -1,58 +1,88 @@
 //
 //  GADMAdapterAppLovinUtils.m
-//  AdMobAdapterDev
 //
-//  Created by Josh Gleeson on 8/15/17.
-//  Copyright © 2017 AppLovin. All rights reserved.
+//
+//  Created by Thomas So on 1/10/18.
+//
 //
 
 #import "GADMAdapterAppLovinUtils.h"
-#import "GADMAdapterAppLovinConstants.h"
+#import "GADMAdapterAppLovinConstant.h"
 #import "GADMAdapterAppLovinExtras.h"
+#import <AppLovinSDK/AppLovinSDK.h>
 
-#if __has_include(<AppLovinSDK/AppLovinSDK.h>)
-    #import <AppLovinSDK/AppLovinSDK.h>
-#else
-    #import "ALSdk.h"
-    #import "ALErrorCodes.h"
-#endif
+#define DEFAULT_ZONE @""
 
 @implementation GADMAdapterAppLovinUtils
 
-+ (ALSdk *)sdkForCredentials:(NSDictionary *)credentials {
-  NSString *sdkKey = [[credentials objectForKey:kGADMAdapterAppLovinSdkKey] copy];
-
-  // if no sdk key pulled from the dashboard, grab the key from the .plist
-  if (sdkKey == nil || [sdkKey isEqualToString:@""]) {
-    NSDictionary *info = [[NSBundle mainBundle] infoDictionary];
-    sdkKey = [info objectForKey:@"AppLovinSdkKey"];
-  }
-
-  ALSdk *sdk = [ALSdk sharedWithKey:sdkKey];
++ (nullable ALSdk *)retrieveSDKFromCredentials:(NSDictionary *)credentials
+{
+    NSString *sdkKey = credentials[GADMAdapterAppLovinConstant.sdkKey];
+    
+    if ( sdkKey.length == 0 )
+    {
+        sdkKey = [[NSBundle mainBundle] infoDictionary][@"AppLovinSdkKey"];
+    }
+    
+    ALSdk *sdk = [ALSdk sharedWithKey: sdkKey];
+    [sdk setPluginVersion: GADMAdapterAppLovinConstant.adapterVersion];
+    
+    return sdk;
+}
 
-  if (sdk) {
-    [sdk setPluginVersion:kGADMAdapterAppLovinVersion];
-  }
++ (NSString *)retrievePlacementFromConnector:(id<GADMediationAdRequest>)connector
+{
+    return connector.credentials[GADMAdapterAppLovinConstant.placementKey] ?: @"";
+}
 
-  return sdk;
++ (NSString *)retrieveZoneIdentifierFromConnector:(id<GADMediationAdRequest>)connector
+{
+    return ((GADMAdapterAppLovinExtras *) connector.networkExtras).zoneIdentifier ?: DEFAULT_ZONE;
```

Comment:
Since the Zone ID has been added to the AdMob FE, you can retrieve it from the connector's credentials using the key: `zone_id`; similar to how the placement is being retrieved above.